modules/tensor_mechanics/src/materials/ComputeCreepPlasticityStress.C

```diff
@@ -207,30 +250,46 @@ ComputeCreepPlasticityStress::updateQpState(RankTwoTensor & elastic_strain_incre
-                 << " plasticity residual = " << plasticity_residual << "\n"
-                 << " creep iteration increment = " << x1 << "\n"
-                 << " plasticity iteration increment = " << x2 << "\n"
-                 << " creep scalar strain = " << effective_creep_strain_increment << "\n"
-                 << " plasticity scalar strain = " << effective_plastic_strain_increment << "\n"
-                 << " stress convergence absolute tolerance = " << _absolute_tolerance << std::endl;
+        _console << "stress iteration number = " << counter << "\n relative residual = "
+                 << (0 == reference_residual ? 0 : residual / reference_residual)
+                 << "\n stress convergence relative tolerance = " << _relative_tolerance
+                 << "\n absolute residual = " << residual
+                 << "\n creep residual = " << creep_residual
+                 << "\n plasticity residual = " << plasticity_residual
+                 << "\n creep iteration increment = " << x1
+                 << "\n plasticity iteration increment = " << x2
+                 << "\n creep scalar strain = " << effective_creep_strain_increment
+                 << "\n plasticity scalar strain = " << effective_plastic_strain_increment
+                 << "\n max creep scalar strain = " << max_effective_creep_strain_increment
+                 << "\n max plasticity scalar strain = " << max_effective_plastic_strain_increment
+                 << "\n stress convergence absolute tolerance = " << _absolute_tolerance
+                 << std::endl;
       }
       ++counter;
     } while (counter < _max_iterations && residual > _absolute_tolerance &&
              (residual / reference_residual) > _relative_tolerance);
 
     if (counter == _max_iterations && residual > _absolute_tolerance &&
         (residual / reference_residual) > _relative_tolerance)
-      throw MooseException("Max stress iteration hit during ComputeCreepPlasticityStress solve!");
+    {
+      std::stringstream msg;
+      msg << "\n relative residual = "
+          << (0 == reference_residual ? 0 : residual / reference_residual)
+          << "\n stress convergence relative tolerance = " << _relative_tolerance
+          << "\n absolute residual = " << residual << "\n creep residual = " << creep_residual
+          << "\n plasticity residual = " << plasticity_residual
+          << "\n creep scalar strain = " << effective_creep_strain_increment
+          << "\n plasticity scalar strain = " << effective_plastic_strain_increment
+          << "\n max creep scalar strain = " << max_effective_creep_strain_increment
+          << "\n max plasticity scalar strain = " << max_effective_plastic_strain_increment
+          << "\n stress convergence absolute tolerance = " << _absolute_tolerance << std::endl;
+
+      throw MooseException("Max stress iteration hit during ComputeCreepPlasticityStress solve! " +
```

Comment:
Is this exception caught properly? I assume it should be with the existing code.

Reply:
Yes, it's caught.  The result is cutting the timestep and trying again.